src/main/java/md/miller1995/Dealership/configs/JWTAuthFilter.java

```diff
@@ -0,0 +1,66 @@
+package md.miller1995.Dealership.configs;
+
+import com.auth0.jwt.exceptions.JWTVerificationException;
+import jakarta.servlet.FilterChain;
+import jakarta.servlet.ServletException;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+import md.miller1995.Dealership.securities.JWTUtil;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.lang.NonNull;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.stereotype.Component;
+import org.springframework.web.filter.OncePerRequestFilter;
+
+import java.io.IOException;
+
+@Component
+public class JWTAuthFilter extends OncePerRequestFilter {
+
+    private final JWTUtil jwtUtil;
+    private final UserDetailsService userDetailsService;
+
+    @Autowired
+    public JWTAuthFilter(JWTUtil jwtUtil, UserDetailsService userDetailsService) {
+        this.jwtUtil = jwtUtil;
+        this.userDetailsService = userDetailsService;
+    }
+
+    @Override
+    protected void doFilterInternal( @NonNull HttpServletRequest request,
+                                     @NonNull HttpServletResponse response,
+                                     @NonNull FilterChain filterChain) throws ServletException, IOException {
+
+        final String authHeader = request.getHeader("Authorization");
+        final String jwt;
+
+        if (authHeader != null && !authHeader.isBlank() && authHeader.startsWith("Bearer ")){
+            jwt = authHeader.substring(7);
+
+            if (jwt.isBlank()){
+                response.sendError(HttpServletResponse.SC_BAD_REQUEST, "Invalid JWT Token in Bearer Header");
```

Comment:
As advice, you should not specify too many details about an exception, because this can lead to important leakage of information to a potential attacker. 
`Invalid JWT Token` I would say it's more than enough as a message exception.